src/components/magnet/magnet.tsx

```diff
@@ -26,23 +32,69 @@ export default class Magnet extends React.Component<IProps, IState> {
   }
 
   public render() {
-    const { draggable, flip, type, image } = this.props;
+    const { draggable, flip, type, image, voltageFlip, imageOffset } = this.props;
     const { x, y } = this.props.magnet;
     // react-pixi typing issue
     const anchor = [0.5, 0.5] as unknown as ObservablePoint;
-    const flipFactor = flip && type === "bar" ? -1 : 1;
-    const scale = [.5 * flipFactor, .5 * flipFactor] as unknown as ObservablePoint;
+    const magFlipFactor = flip && type === "bar" ? -1 : 1;
+    const magScale = [.5 * magFlipFactor, .5 * magFlipFactor] as unknown as ObservablePoint;
+    const voltFlipFactor = voltageFlip && type === "coil" ? -1 : 1;
+    const voltScale = [.5 * voltFlipFactor, .5] as unknown as ObservablePoint;
+    const scale = [.5, .5] as unknown as ObservablePoint;
+    const leftCurrScale = [.5, .5 * voltFlipFactor] as unknown as ObservablePoint;
+    const rightCurrScale = [.5, .5 * -1 * voltFlipFactor] as unknown as ObservablePoint;
     return (
-      <Sprite
-        image={image} x={x} y={y}
-        scale={scale}
-        anchor={anchor}
-        interactive={draggable ? true : false}
-        buttonMode={draggable ? true : false}
-        pointerdown={draggable ? this.onPointerDown : undefined}
-        pointerup={draggable ? this.onPointerUp : undefined}
-        pointermove={draggable ? this.onPointerMove : undefined}
-      />
+      <Container>
+        <Sprite
+          image={image} x={x} y={y}
+          scale={magScale}
+          anchor={anchor}
+          interactive={draggable ? true : false}
+          buttonMode={draggable ? true : false}
+          pointerdown={draggable ? this.onPointerDown : undefined}
+          pointerup={draggable ? this.onPointerUp : undefined}
+          pointermove={draggable ? this.onPointerMove : undefined}
+        />
+        { this.props.voltageImage !== ""
+          ? <Sprite
+              image={this.props.voltageImage} x={x - (voltFlipFactor * 28)} y={y + 58}
+              scale={voltScale}
+            />
+          : null
+        }
+        { this.props.leftPoleImage !== ""
+          ? <Sprite
+              image={this.props.leftPoleImage} x={x - (66 + imageOffset)} y={y}
+              scale={scale}
+              anchor={anchor}
+            />
+          : null
+        }
+        { this.props.rightPoleImage !== ""
+          ? <Sprite
+              image={this.props.rightPoleImage} x={x + (66 + imageOffset)} y={y}
+              scale={scale}
+              anchor={anchor}
+            />
+          : null
+        }
+        { this.props.currentArrowImage !== ""
+          ? <Sprite
+              image={this.props.currentArrowImage} x={x - (66 + imageOffset)} y={y + 50}
+              scale={leftCurrScale}
+              anchor={anchor}
+            />
+          : null
+        }
+        { this.props.currentArrowImage !== ""
+          ? <Sprite
+              image={this.props.currentArrowImage} x={x + (66 + imageOffset)} y={(y + 50)}
```

Comment:
Would be nice to break out these constants too

Reply:
Agreed.  Will do.

Reply:
Added these values as constants.